File: EncoraTest/ViewModels/property.ts
import { Address } from './address'
import { Financial } from './financial'
import { Physical } from './physical'
import { Score } from './score'
import { Valuation } from './valuation'
import { Resource } from './resource'
import { Lease } from './lease'
import { Diligence } from './diligence'
import { GoogleMapOption } from './googleMapOption'

export interface Property {
    id: number;
    accountId: number;
    buyerAccountId: number;
    externalId: number;
    programId: number;
    isDwellCertified: boolean;
    isAllowOffer: boolean;
    isAllowPreview: boolean;
    isFeature: boolean;
    isRentGuaranteed: boolean;
    allowRentGuaranteedOptout: boolean;
    isSecuritized: boolean;
    isHot: boolean;
    isNew: boolean;
    isBargain: boolean;
    isDiligenceVaultUnlocked: boolean;
    isPropertyManagerOfferRetain: boolean;
    isHoa: boolean;
    hasAudio: boolean;
    hasDiligenceVaultDocuments: boolean;
    marketId: number;
    daysOnMarket: number;
    latitude: number;
    longitude: number;
    name: string;
    description: string;
    highlights: string;
    mainImageUrl: string;
    personalProperties: string;
    diligenceVaultSummary: string;
    featuredReason: string;
    status: string;
    allowedFundingTypes: string;
    allowableSaleTypes: string;
    visibility: string;
    priceVisibility: string;
    type: string;
    certificationLevel: string;
    escrowClosingDate: string;
    address: Address;
    financial: Financial;
    physical: Physical;
    score: Score;
    valuation: Valuation;
    resource: Resource;
    manager: string;
    seller: string;
    sellerBroker: string;
    hoa: string;
    lease: Lease;
    diligences: Diligence[];
    googleMapOption: GoogleMapOption;
    inspectionType: string;
}

File: EncoraTestBLL/Models/Property.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EncoraTestBLL.Models
{
    public class Property
    {

        #region class properties

        [DatabaseGenerated(DatabaseGeneratedOption.None)]
        [Key]
        public int? Id { get; set; }

        public int? AccountId { get; set; }

        public int? BuyerAccountId { get; set; }

        public int? ExternalId { get; set; }

        public int? ProgramId { get; set; }

        public bool? IsDwellCertified { get; set; }

        public bool? IsAllowOffer { get; set; }

        public bool? IsAllowPreview { get; set; }

        public bool? IsFeature { get; set; }

        public bool? IsRentGuaranteed { get; set; }

        public bool? AllowRentGuaranteedOptout { get; set; }

        public bool? IsSecuritized { get; set; }

        public bool? IsHot { get; set; }

        public bool? IsNew { get; set; }

        public bool? IsBargain { get; set; }

        public bool? IsDiligenceVaultUnlocked { get; set; }

        public bool? IsPropertyManagerOfferRetain { get; set; }

        public bool? IsHoa { get; set; }

        public bool? HasAudio { get; set; }

        public bool? HasDiligenceVaultDocuments { get; set; }

        public int? MarketId { get; set; }

        public int? DaysOnMarket { get; set; }

        public long? Latitude { get; set; }

        public long? Longitude { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public string Highlights { get; set; }

        public string MainImageUrl { get; set; }

        public string PersonalProperties { get; set; }

        public string DiligenceVaultSummary { get; set; }

        public string FeaturedReason { get; set; }

        public string Status { get; set; }

        public string AllowedFundingTypes { get; set; }

        public string AllowableSaleTypes { get; set; }

        public string Visibility { get; set; }

        public string PriceVisibility { get; set; }

        public string Type { get; set; }

        public string CertificationLevel { get; set; }

        public DateTime? EscrowClosingDate { get; set; }

        public Address Address { get; set; }

        public Financial Financial { get; set; }

        public Physical Physical { get; set; }

        public Score Score { get; set; }

        public Valuation Valuation { get; set; }

        public Resource Resource { get; set; }

        public string Manager { get; set; }

        public string Seller { get; set; }

        public string SellerBroker { get; set; }

        public string Hoa { get; set; }

        public Lease Lease { get; set; }

        public List<Diligence> Diligences { get; set; }

        public GoogleMapOption GoogleMapOption { get; set; }

        public string InspectionType { get; set; }

     

        #endregion



        public Property()
        {
        }
    }
}
